### PR TITLE
Add documentation for event handlers

### PR DESCRIPTION
**Pull request by mmghannam**

### CODE REVIEW

File: docs/tutorials/eventhandler.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+.. contents:: Contents
+
+SCIP Events
+===========
+SCIP provides a number of events that can be used to interact with the solver. These events would describe a change in the model state before or during the solving process; For example a variable/constraint were added or a new best solution is found.
+The enum :code:`pyscipopt.SCIP_EVENTTYPE` provides a list of all available events.
+
+
+Adding Event Handlers
+==============
```
Comment (Opt-Mucca) [inline docs/tutorials/eventhandler.rst:22]: I'm never sure on this, but I think you should have at least as many "=" characters as there are in the section titles

**Review by Opt-Mucca — CHANGES_REQUESTED** [review]
The only change I'd suggest is splitting the second subsection into multiple subsections. Something like:
"What is an event handler?"
"Adding Event Handlers with Callbacks"
"Add Event Handler Classes"

**Comment by Opt-Mucca** [comment]
I am super happy to see you already building on the documentation! 
I'm a bit picky on this one because I don't see us changing stuff regularly here once it's added here.

File: docs/tutorials/eventhandler.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+
+The callback function should have the following signature: :code:`def callback(model, event)`.
+The first argument is the model object and the second argument is the event that occurred.
+
+
+If you need to store additional data in the event handler, you can create a custom event handler class that inherits from :code:`pyscipopt.Eventhdlr`.
+and then include it in the model using the :code:`Model.includeEventHandler` method. The following is an example that stores the number of best solutions found:
+
+.. code-block:: python
```
[the remark this answers is not in the document]
Comment (mmghannam) [reply docs/tutorials/eventhandler.rst:57]: yup you're right, done!

**Comment by mmghannam** [comment]
> I am super happy to see you already building on the documentation!

You've made it quite smooth to add on the structure you've built so thank you! 

> I'm a bit picky on this one because I don't see us changing stuff regularly here once it's added here.

No worries, pick all you want 😄 anything that's going to make it a better experience for users I'm all for, and thanks for the quick review!

**Review by Opt-Mucca — APPROVED** [review]

**Comment by Opt-Mucca** [comment]
We have liftoff!

**Pull request merged**